core/pkg/otel/metrics_test.go

```diff
@@ -90,65 +90,84 @@ func TestNewOTelRecorder(t *testing.T) {
 }
 
 func TestMetrics(t *testing.T) {
-	exp := metric.NewManualReader()
-	rec := NewOTelRecorder(exp, svcName)
-	ctx := context.TODO()
 	attrs := []attribute.KeyValue{
 		semconv.ServiceNameKey.String(svcName),
 	}
 	const n = 5
-	type MetricF func()
+	type MetricF func(exp metric.Reader)
 	tests := []struct {
 		name       string
 		metricFunc MetricF
+		metricsLen int
 	}{
 		{
 			name: "HTTPRequestDuration",
-			metricFunc: func() {
+			metricFunc: func(exp metric.Reader) {
+				rec := NewOTelRecorder(exp, svcName)
 				for i := 0; i < n; i++ {
-					rec.HTTPRequestDuration(ctx, 10, attrs)
+					rec.HTTPRequestDuration(context.TODO(), 10, attrs)
 				}
 			},
+			metricsLen: 1,
 		},
 		{
 			name: "HTTPResponseSize",
-			metricFunc: func() {
+			metricFunc: func(exp metric.Reader) {
+				rec := NewOTelRecorder(exp, svcName)
 				for i := 0; i < n; i++ {
-					rec.HTTPResponseSize(ctx, 100, attrs)
+					rec.HTTPResponseSize(context.TODO(), 100, attrs)
 				}
 			},
+			metricsLen: 1,
 		},
 		{
 			name: "InFlightRequestStart",
-			metricFunc: func() {
+			metricFunc: func(exp metric.Reader) {
+				rec := NewOTelRecorder(exp, svcName)
+				ctx := context.TODO()
 				for i := 0; i < n; i++ {
 					rec.InFlightRequestStart(ctx, attrs)
 					rec.InFlightRequestEnd(ctx, attrs)
 				}
 			},
+			metricsLen: 1,
 		},
 		{
 			name: "Impressions",
-			metricFunc: func() {
+			metricFunc: func(exp metric.Reader) {
+				rec := NewOTelRecorder(exp, svcName)
+				for i := 0; i < n; i++ {
+					rec.Impressions(context.TODO(), "reason", "variant", "key")
+				}
+			},
+			metricsLen: 1,
+		},
+		{
+			name: "Reasons",
+			metricFunc: func(exp metric.Reader) {
+				rec := NewOTelRecorder(exp, svcName)
 				for i := 0; i < n; i++ {
-					rec.Impressions(ctx, "key", "variant")
+					rec.Reasons(context.TODO(), "reason")
 				}
 			},
+			metricsLen: 1,
 		},
```

Comment:
I would also add 2 entries for `RecordEvaluation`: one with and one without error